new/missing-method-missing.md

```diff
@@ -0,0 +1,156 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Remove magic and method_missing from Chef::Resource lookup
+
+This proposal:
+
+- Switches to explicit methods for Chef recipe DSL instead of method_missing
+- Removes all special treatment of classes in the Chef::Resource and Chef::Provider namespaces
+- Automatically adds resource DSL for all named descendants of Chef::Resource, no matter what their namespace
+
+## Motivation
+
+    As a Chef user,
+    I want resources declared in my own namespace to be in recipe DSL automatically,
+    so that I don't have to write the name of the resource twice.
+
+    As a Chef user,
+    I want to declare resources in my own namespace,
+    So that I get better error messages when there is a resource conflict.
+
+    As a Chef developer,
+    I want meaningful names in stack traces for methods,
+    So that I can actually read the debug output.
+
+    As a Chef developer,
+    I want less if statements, case statements and metaprogramming in DSL method calls,
+    So that I can debug and follow logic more easily.
+
+## Specification
+
+### Create explicit DSL methods for resources
+
+When looking up resources, there is no method anywhere that corresponds to the
+actual resource name.  This can lead to a lot of unnecessarily heavy thinking
+when following some very typical stack traces.  It also means that the name of
+the resource being declared is often not on the stack, making debugging harder.
+
+Here we propose that all resource and definition DSL be added to
```

Comment:
I use it for running tests, anonymous resources are created before the example runs and removed afterwards. Otherwise I would end up declaring resources like `poise_test48` all over my tests to avoid name collisions and that gets annoying.